test/dataset/split/test_split.py

```diff
@@ -39,50 +40,47 @@ def test_ts_slice_to_item():
     sl.to_data_entry()
 
 
-def test_splitter():
-    dataset = get_dataset("m4_hourly")
-    prediction_length = dataset.metadata.prediction_length
-    splitter = DateSplitter(
-        prediction_length=prediction_length,
-        split_date=pd.Period("1750-01-05 04:00:00", freq="h"),
+def check_training_validation(
+    train_entry,
+    valid_pair,
+    prediction_length: int,
+    max_history,
+    date,
+    offset,
+) -> None:
+    assert train_entry[FieldName.ITEM_ID] == valid_pair[0][FieldName.ITEM_ID]
+    if max_history is None:
+        assert len(train_entry[FieldName.TARGET]) == len(
+            valid_pair[0][FieldName.TARGET]
+        )
+    else:
```

Comment:
I think the first case needs not to be checked, so we can simply do

```suggestion
    if max_history is not None:
```